engine/classes/Elgg/Database/EntityTable.php

```diff
@@ -140,13 +141,16 @@ function rowToElggStar($row) {
 	/**
 	 * Loads and returns an entity object from a guid.
 	 *
-	 * @param int    $guid The GUID of the entity
-	 * @param string $type The type of the entity. If given, even an existing entity with the given GUID
-	 *                     will not be returned unless its type matches.
+	 * @param int    $guid             Entity GUID
+	 * @param string $type             Entity type. If given, even an existing entity with the given GUID
+	 *                                 will not be returned unless its type matches.
+	 * @param string $subtype          Entity subtype (use "" for default subtype). If given, even an existing
+	 *                                 entity with the given GUID will not be returned unless this matches.
+	 * @param bool   $ignore_condition If true, hidden/disabled entities may be found
 	 *
 	 * @return \ElggEntity The correct Elgg or custom object based upon entity type and subtype
 	 */
-	function get($guid, $type = '') {
+	function get($guid, $type = '', $subtype = null, $ignore_condition = false) {
```

Comment:
IMHO, it would server a shortcut to getting an entity of a specific type and subtype. Would eliminate the need to validate the subtype on an entity.